### DISCUSSION TITLE
BSD 3-Clause License @ mobile app

### DISCUSSION THREAD

**Discussion by kkondarev** [rydmike/flex_color_picker#59] [Q&A] [answered]
Hi, 

I would like to use the color picker in a mobile app. Looking at the license agreement, specifically point 2:
"Redistributions in binary form must reproduce the above copyright notice,
   this list of conditions and the following disclaimer in the documentation
   and/or other materials provided with the distribution."

There is no documentation or anything else shipped with the application. Would it be sufficient to mention the color picker on the accompanying web site in my attribution section and link to the license? If not, would would be your advice to proceed in case of Android/iOS mobile app?

Thanks and best regards
Kras

**Comment by rydmike** [comment, marked as answer]
Hi @kkondarev thanks for reaching out and asking.

Since you are making a Flutter app that depends on a large amount amount of open source libraries, with the similar license terms terms to include the copyright notice in your redistributed app. I recommend that you in your app's about section/dialog, use the standard **Flutter About** dialog feature, to show the version info for you app and other things you want to say about it. This dialog contains a button to "View Licenses" that can correctly list and display **all** open source components and licenses your application uses. This nicely takes care of all your legal requirements to disclose used open source libraries in your application.

Flutter framework itself is BSD-3 clause as well by the way, it just does not mention the name BSD-3 by name anymore, I think it did earlier, but text is identical.
 
When you use the show licenses via the about dialog, you will be surprised to see how many licenses that there are in your app that you should/must list and display license info for. Using this dialog takes care of them **all** in in automatic manner for you. You can of course re-uses the code from the About dialog that shows the licenses to put it it elsewhere, in some other appropriate place in your app, than in an about dialog, but I have found it works well enough as is.

Below an example, the built in "View licenses" ui works fine in a phone and in a desktop app as can be seen below.

![image](https://user-images.githubusercontent.com/39990307/158025470-14bdd95c-3a1a-4d70-b24e-687fa33cd10e.png)

![image](https://user-images.githubusercontent.com/39990307/158025474-b0c562c4-e4a3-4393-8566-69dbb9ecf023.png)

![image](https://user-images.githubusercontent.com/39990307/158025481-9c3dacbd-6b8d-4cf5-8b1b-296cca95f8c0.png)

![image](https://user-images.githubusercontent.com/39990307/158025482-dccca3e2-d77b-4825-8258-a10eed90ce6d.png)

Even the FlexColorPicker web demo uses this dialog:

![image](https://user-images.githubusercontent.com/39990307/158026146-b8418410-ea48-4917-8671-ce5a6cec5475.png)

![image](https://user-images.githubusercontent.com/39990307/158026165-02328956-f3a4-4bf0-88b8-d70ac4c8103b.png)


If you need code example on how to use the About Dialog, then FlexColorPicker web demo itself is open source and bundled with the package repo.

Hope this helps and answers your question 💙 

BR Mike

**Comment by kkondarev** [comment]
Hi Mike,

thank you for the very detailed answer!
I will implement the About Dialog as suggested. Thanks!

Best
Kras
